### PR TITLE
Add VERBOSE=1 env variable to our build and test scripts

### PR DESCRIPTION
**Pull request by arcusfelis**
This PR addresses "ugly makefiles" and "not enough info in the shell sometimes".

Proposed changes include:
* Move pipe magic from makefiles into `tools/silent_exec.sh`
* Allow to disable pipe magic, if `VERBOSE=1` env variable is set.
* Silencing cover in `travis-test.sh`.
* Fix `ln: illegal option -- d` on Mac, thanks @fenek

### CODE REVIEW

**Comment by mongoose-im** [comment]
[4907.5](https://travis-ci.org/esl/MongooseIM/jobs/391749043) / Erlang 19.3 / elasticsearch_and_cassandra_mnesia / 6438e6477d0ea819fd2f74b3741ac3d574681fa7
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/1926/4907/elasticsearch_and_cassandra_mnesia.19.3) / [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1926/4907/elasticsearch_and_cassandra_mnesia.19.3/big/ct_run.test%40travis-job-f5c53bce-8860-4675-a0e4-7f53be9a0da2.2018-06-13_13.31.31/index.html) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1926/4907/elasticsearch_and_cassandra_mnesia.19.3/small/ct_run.mongooseim@localhost.2018-06-13_13.24.16/lib.mongooseim.logs/run.2018-06-13_13.24.16/suite.log.html)
**OK: 445** / Failed: 0 / **User-skipped: 8** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[4909.5](https://travis-ci.org/esl/MongooseIM/jobs/391811751) / Erlang 19.3 / elasticsearch_and_cassandra_mnesia / 7ac8cd8c5c883cc9261043f7740c4739c7d262c9
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/1926/4909/elasticsearch_and_cassandra_mnesia.19.3) / [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1926/4909/elasticsearch_and_cassandra_mnesia.19.3/big/ct_run.test%40travis-job-30650b78-ed88-4450-bd2c-76f166944b74.2018-06-13_15.45.52/index.html) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1926/4909/elasticsearch_and_cassandra_mnesia.19.3/small/ct_run.mongooseim@localhost.2018-06-13_15.38.40/lib.mongooseim.logs/run.2018-06-13_15.38.41/suite.log.html)
**OK: 445** / Failed: 0 / **User-skipped: 8** / Auto-skipped: 0

File: big_tests/run_common_test.erl [hunk 1]
```diff
@@ -588,3 +590,84 @@ handle_file_error(_FileName, Other) ->
...
+%% cover_server process is using io:format too much.
+%% This code removes duplicate io:formats.
+%% Each io:format would be written only once with this group leader.
+%%
+%% Example of a message we want to write only once:
+%% "Analysis includes data from imported files" from cover.erl in Erlang/R19
+%% ------------------------------------------------------------------
+
+%% Set a new group leader for cover_server
+deduplicate_cover_server_console_prints() ->
```
Comment (michalwski) [inline big_tests/run_common_test.erl]: I'd like to see this functionality moved to a separate module.

Comment (arcusfelis) [reply big_tests/run_common_test.erl]: separated.

**Comment by arcusfelis** [comment]
"ugly makefiles" step 1/100.

**Comment by mongoose-im** [comment]
[4915.4](https://travis-ci.org/esl/MongooseIM/jobs/392141039) / Erlang 19.3 / ldap_mnesia / 72c8e78694d326be91eb96ccbccbb748529a1ceb
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/1926/4915/ldap_mnesia.19.3) / [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1926/4915/ldap_mnesia.19.3/big/ct_run.test%40travis-job-ad439d76-5a44-416b-aa37-36b6f36e4fe5.2018-06-14_08.45.33/index.html) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1926/4915/ldap_mnesia.19.3/small/ct_run.mongooseim@localhost.2018-06-14_08.38.42/lib.mongooseim.logs/run.2018-06-14_08.38.43/suite.log.html)
**OK: 1032** / Failed: 0 / **User-skipped: 77** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[4916.1](https://travis-ci.org/esl/MongooseIM/jobs/392141991) / Erlang 19.3 / internal_mnesia / 22421a863211149e16861d804f79fe469fa26fad
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/1926/4916/internal_mnesia.19.3) / [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1926/4916/internal_mnesia.19.3/big/ct_run.test%40travis-job-2030ed3b-3963-4328-ab74-aea38d31f1a3.2018-06-14_09.21.38/index.html) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/1926/4916/internal_mnesia.19.3/small/ct_run.mongooseim@localhost.2018-06-14_09.14.34/lib.mongooseim.logs/run.2018-06-14_09.14.34/suite.log.html)
**OK: 1068** / Failed: 0 / **User-skipped: 41** / Auto-skipped: 0

**Pull request merged**